lib/Centreon/Object/Relation/Host/Template/Host.php

```diff
@@ -70,6 +70,46 @@ public function insert($fkey, $skey = null)
         $this->db->query($sql, array($fkey, $skey, $order));
     }
 
+    /**
+     * Delete host template / host relation and linked services
+     *
+     * @param int $fkey
+     * @param int $skey
+     * @return void
+     */
+    public function delete($fkey, $skey = null): void
+    {
+        $this->db->beginTransaction();
+        try {
+            parent::delete($fkey, $skey);
+            // Delete linked services :
+            // First get service IDs
+            $stmt = $this->db->prepare(
+                'SELECT service_service_id FROM host_service_relation WHERE host_host_id = :host_id'
+            );
+            $stmt->bindParam(':host_id', $fkey, PDO::PARAM_INT);
+            $stmt->execute();
+            $services = [];
+            foreach ($stmt->fetchAll(PDO::FETCH_ASSOC) as $row) {
+                $services[] = $row['service_service_id'];
+            }
+
+            if (!empty($services)) {
+                // If there are linked services - delete these services
+                $stmt = $this->db->prepare('DELETE FROM host_service_relation 
+                             WHERE host_host_id = :host_id
+                             AND service_service_id IN (:services)');
```

Comment:
I guess it does not work...

Reply:
Yes, it does not work, fixed.

Reply:
I think so too.
https://stackoverflow.com/questions/1586587/pdo-binding-values-for-mysql-in-statement